src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -132,12 +133,10 @@ private Task OnProjectFactoryCompletedAsync()
         {
             var previousProjectContext = _currentAggregateProjectContext;
             var newProjectContext = await UpdateProjectContextAsync().ConfigureAwait(false);
+
             if (previousProjectContext != newProjectContext)
             {
-                // Dispose existing subscriptions.
-                DisposeAndClearSubscriptions();
```

Comment:
In NuGetRestorer we ignore updates that do not contain any changes, here: [ProjectRestoreInfoBuilder.cs#L33](https://github.com/dotnet/roslyn-project-system/blob/637bfdf74ccf9ab2de20556e43cd54d48904a691/src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/ProjectRestoreInfoBuilder.cs#L33). Also, NuGet's solution restorer is robust to handle multiple restore calls with identical information. I'm not sure about the way we are disposing -- we use ProjectDataSources.SyncLinkTo to create one subscription which is disposed when target framework changes. We can revisit this when we investigate https://github.com/dotnet/roslyn-project-system/issues/1503